packages/stream/src/middleware.ts

```diff
@@ -1,66 +1,70 @@
 import { defineMiddleware } from "astro:middleware";
 
 type SuspendedChunk = {
-	chunk: string;
-	idx: number;
+  chunk: string;
+  idx: number;
 };
 
 export const onRequest = defineMiddleware(async (ctx, next) => {
-	const response = await next();
-	// ignore non-HTML responses
-	if (!response.headers.get("content-type")?.startsWith("text/html")) {
-		return response;
-	}
+  let streamController: ReadableStreamDefaultController<SuspendedChunk>;
 
-	let streamController: ReadableStreamDefaultController<SuspendedChunk>;
+  // Thank you owoce!
+  // https://gist.github.com/lubieowoce/05a4cb2e8cd252787b54b7c8a41f09fc
+  const stream = new ReadableStream<SuspendedChunk>({
+    start(controller) {
+      streamController = controller;
+    },
+  });
 
-	async function* render() {
-		// Thank you owoce!
-		// https://gist.github.com/lubieowoce/05a4cb2e8cd252787b54b7c8a41f09fc
-		const stream = new ReadableStream<SuspendedChunk>({
-			start(controller) {
-				streamController = controller;
-			},
-		});
+  let curId = 0;
+  const pending = new Set<Promise<string>>();
 
-		let curId = 0;
-		const pending = new Set<Promise<string>>();
+  ctx.locals.suspend = (promise) => {
```

Comment:
same as the other PR -- pulled this out to the beginning of the middleware to init `Astron.locals.suspend` ASAP to avoid any weird ordering issues.